core/src/commands/workflow.ts

```diff
@@ -205,14 +207,15 @@ export class WorkflowCommand extends Command<Args, {}> {
         logErrors(outerLog, stepResult.errors, index, steps.length, step.description)
         stepErrors[index] = stepResult.errors
         // There may be succeeding steps with `when: onError` or `when: always`, so we continue.
+        shouldError = shouldError || !step.allowFailure
```

Comment:
The comment was meant for the `continue` line, so let's keep them together
```suggestion
        shouldError = shouldError || !step.allowFailure
        // There may be succeeding steps with `when: onError` or `when: always`, so we continue.
```